contracts/ccip/GovernanceCCIPRelay.sol

```diff
@@ -32,33 +31,112 @@ contract GovernanceCCIPRelay is IGovernanceCCIPRelay {
   /// @param _timelock The address of the Timelock contract.
   /// @param _router The address of the CCIP router contract.
-  /// @param _destinationChainSelector The chain selector for the destination chain.
-  /// @param _receiver The address of the receiver contract on the destination chain.
+  /// @param _destinationChainSelectors Array of chain selectors for the destination chains.
+  /// @param _destinationReceivers Array addresses of the receiver contracts on the destination chains.
   constructor(
     address _timelock,
     address _router,
-    uint64 _destinationChainSelector,
-    address _receiver
+    uint64[] memory _destinationChainSelectors,
+    address[] memory _destinationReceivers
   ) {
     ccipRouter = IRouterClient(_router);
     timelock = _timelock;
-    destinationChainSelector = _destinationChainSelector;
-    destinationReceiver = _receiver;
+    _addDestinationChains(_destinationChainSelectors, _destinationReceivers);
   }
 
   /// @inheritdoc IGovernanceCCIPRelay
-  function setDestinationReceiver(address _receiver) external onlyTimeLock {
-    emit DestinationReceiverUpdated(destinationReceiver, _receiver);
-    destinationReceiver = _receiver;
+  function addDestinationChains(
+    uint64[] memory _destinationChainSelectors,
+    address[] memory _destinationReceivers
+  ) external onlyTimeLock {
+    _addDestinationChains(_destinationChainSelectors, _destinationReceivers);
+  }
+
+  function _addDestinationChains(
+    uint64[] memory _destinationChainSelectors,
+    address[] memory _destinationReceivers
+  ) private {
+    uint256 receiversLength = _destinationReceivers.length;
+    require(
+      _destinationChainSelectors.length == receiversLength,
+      ArrayLengthMismatch()
+    );
+
+    for (uint256 i = 0; i < receiversLength; ) {
+      uint64 _destinationChainSelector = _destinationChainSelectors[i];
+      address _destinationReceiver = _destinationReceivers[i];
+
+      require(
+        ccipRouter.isChainSupported(_destinationChainSelector),
+        DestinationChainNotSupported(_destinationChainSelector)
+      );
+
+      require(
+        _destinationChainSelector != CCIP_MAINNET_CHAIN_SELECTOR,
+        CannotUseMainnetChainSelector()
+      );
+
+      require(
+        _destinationReceiver != address(0),
+        ReceiverCannotBeZeroAddress()
+      );
+
+      require(
+        destinationReceivers[_destinationChainSelector] == address(0),
+        ChainSelectorAlreadyAssigned(_destinationChainSelector)
+      );
+
+      destinationReceivers[_destinationChainSelector] = _destinationReceiver;
+      emit DestinationChainAdded(
+        _destinationChainSelector,
+        _destinationReceiver
+      );
+
+      unchecked {
+        i++;
+      }
+    }
   }
 
   /// @inheritdoc IGovernanceCCIPRelay
-  function relayMessage(address target, bytes calldata payload)
-    external
-    payable
-    onlyTimeLock
-    returns (bytes32 messageId)
-  {
+  function updateDestinationReceiver(
```

Comment:
@voith this function correctly update the receiver for a `_destinationChainSelector` but **does not** allow the timelocker to **remove** support for a `_destinationChainSelector` given that you are reverting if `_destinationReceiver == address(0)`.

Do you want to support this scenario where you "deprecate" a `_destinationChainSelector`?

Reply:
to be clear, Chainlink **could** deprecate it on the router.

Reply:
I understand. Leaving deprecated data inside a mapping doesn't worry too much as I don't see a huge potential issue or an exploit happening with it.